### PR TITLE
Adding builtins

### PR DESCRIPTION
**Pull request by ChristianoBraga**
Added the following builtins: addInteger, multiplyInteer, subtractInteger, divideInteger, lessThanInteger.

### CODE REVIEW

**Comment by ChristianoBraga** [comment]
The force.uplc test in tests/simple is failing because the semantics changed. This is making jenkins unhappy.

File: tests/failing [hunk 1]
```diff
@@ -1,3 +1,4 @@
+tests/simple/error.uplc
```
Comment (SchmErik) [inline tests/failing:1]: out of curiosity, why is this test listed as a failing test? I see that there is an expected output file and it looks good to me. What happens if this line is removed?

File: uplc.md [hunk 2]
```diff
@@ -1,156 +1,173 @@
...
-   syntax Value ::= "(" "con" TypeConstant Constant ")" // constant
-                  | "(" "lam" Var Term ")"              // lambda abstraction
-                  | "(" "delay" Term ")"                // delay execution of a term
-
-   syntax Term ::= Var
+   imports BOOL
+
+   syntax TypeConstant ::= "integer"
+                         | "data"
+			 | "bytestring"
```
Comment (SchmErik) [inline uplc.md:12]: I see tabs here. Please replace tabs with spaces

File: uplc.md [hunk 3]
```diff
@@ -1,156 +1,173 @@
...
+			 | "unit"
+			 | "bool"
+			 
+   syntax ByteString   ::= r"#([a-fA-F0-9][a-fA-F0-9])+" [token]
+   
+   syntax Constant     ::= Int
+                         | "True"
+			 | "False"
```
Comment (SchmErik) [inline uplc.md:20]: I see tabs here. Please replace tabs with spaces

File: uplc.md [hunk 4]
```diff
@@ -1,156 +1,173 @@
...
+			 | ByteString
+			 | "()"
+			 
+   syntax BuiltinName  ::= "addInteger"
+                         | "multiplyInteger"
+                         | "subtractInteger"
+			 | "divideInteger"
```
Comment (SchmErik) [inline uplc.md:27]: I see tabs here. Please replace tabs with spaces

File: uplc.md [hunk 5]
```diff
@@ -1,156 +1,173 @@
...
+			 | "lessThanInteger"
+			 | "sha3_256"
+
+   syntax Value ::= "(" "con" TypeConstant Constant ")" 
+                  | "(" "lam" Id Term ")"              
+                  | "(" "delay" Term ")"                
+		  | "#SUM"
```
Comment (SchmErik) [inline uplc.md:34]: I see tabs here. Please replace tabs with spaces

File: uplc.md [hunk 6]
```diff
@@ -1,156 +1,173 @@
...
+
+   syntax TermList ::= NeList{Term, ""}
+
+   syntax Term ::= Id
                  | Value 
                  | "[" Term Term "]"                     // function application
                  | "(" "force" Term ")"                  // force execution of a term
-                 | "(" "builtin" BuiltinName List ")"    // builtin
+                 | "(" "builtin" BuiltinName ")"
+		 | "(" "builtin" BuiltinName TermList ")"// builtin
```
Comment (SchmErik) [inline uplc.md:58]: I see tabs here. Please replace tabs with spaces

File: uplc.md [hunk 7]
```diff
@@ -1,156 +1,173 @@
...
-                    | "[" AClosure "_]"
-                    | BuiltinApp(BuiltinName, List, List, Map)
-                    | "Force"
 
-  syntax TypeConstant ::= "integer" [token]
-
-  syntax BuiltinName ::= "addInteger" [token]
+  syntax ATerm ::= "Force"
+                 | "[_" Term "]"
+		 | "[" AClosure "_]"
```
Comment (SchmErik) [inline uplc.md:76]: I see tabs here. Please replace tabs with spaces

**Review by SchmErik — CHANGES_REQUESTED** [review]
Over all looks good! I saw that there were several tabs so please go through uplc.md and fix them. Also, please check out my question regarding the tests/failing file

**Review by SchmErik — APPROVED** [review]
looks good to me

**Pull request merged**